### PR TITLE
We need the xamarin-impl feed.

### PR DESCRIPTION
**Pull request by rolfbjarne**
Otherwise this happens:

    Errors in [...]/xamarin-macios/msbuild/Xamarin.MacDev.Tasks/Xamarin.MacDev.Tasks.csproj
        NU1101: Unable to find package Xamarin.Messaging.Build.Client. No packages exist with this id in source(s): Dotnet arcade, dotnet-public, dotnet6, macios-dependencies

    Errors in [...]/xamarin-macios/msbuild/Xamarin.iOS.Tasks/Xamarin.iOS.Tasks.csproj
        NU1101: Unable to find package Xamarin.Messaging.Build.Client. No packages exist with this id in source(s): Dotnet arcade, dotnet-public, dotnet6, macios-dependencies

    Errors in [...]/xamarin-macios/msbuild/Xamarin.Mac.Tasks/Xamarin.Mac.Tasks.csproj
        NU1101: Unable to find package Xamarin.Messaging.Build.Client. No packages exist with this id in source(s): Dotnet arcade, dotnet-public, dotnet6, macios-dependencies

    Errors in [...]/xamarin-macios/msbuild/Xamarin.iOS.Tasks.Windows/Xamarin.iOS.Tasks.Windows.csproj
        NU1101: Unable to find package Xamarin.Messaging.Build.Client. No packages exist with this id in source(s): Dotnet arcade, dotnet-public, dotnet6, macios-dependencies
        NU1101: Unable to find package Xamarin.Messaging.Core. No packages exist with this id in source(s): Dotnet arcade, dotnet-public, dotnet6, macios-dependencies
        NU1101: Unable to find package Xamarin.iOS.HotRestart.Client. No packages exist with this id in source(s): Dotnet arcade, dotnet-public, dotnet6, macios-dependencies
        NU1101: Unable to find package Xamarin.Messaging.Server. No packages exist with this id in source(s): Dotnet arcade, dotnet-public, dotnet6, macios-dependencies

    Errors in [...]/xamarin-macios/msbuild/Messaging/Xamarin.Messaging.Build/Xamarin.Messaging.Build.csproj
        NU1101: Unable to find package Xamarin.Messaging.Core. No packages exist with this id in source(s): Dotnet arcade, dotnet-public, dotnet6, macios-dependencies

    Errors in [...]/xamarin-macios/msbuild/Xamarin.Mac.Tasks.Core/Xamarin.Mac.Tasks.Core.csproj
        NU1101: Unable to find package Xamarin.Messaging.Build.Client. No packages exist with this id in source(s): Dotnet arcade, dotnet-public, dotnet6, macios-dependencies

It was removed in c272040bdb3aa988c48864183500487c404713b3, so this is a partial revert of that commit.

### CODE REVIEW

**Comment by rolfbjarne** [comment]
CC @pjcollins

File: NuGet.config [hunk 1]
```diff
@@ -11,6 +11,7 @@
     <add key="dotnet5" value="https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet5/nuget/v3/index.json" />
     <add key="xamarin-impl" value="https://pkgs.dev.azure.com/azure-public/vside/_packaging/xamarin-impl/nuget/v3/index.json" />
     <add key="macios-dependencies" value="https://pkgs.dev.azure.com/xamarin/public/_packaging/macios-dependencies/nuget/v3/index.json" />
+    <add key="xamarin-impl" value="https://pkgs.dev.azure.com/azure-public/vside/_packaging/xamarin-impl/nuget/v3/index.json" />
```
Comment (filipnavara) [inline NuGet.config:14]: I am confused, the same source is two lines above

Comment (rolfbjarne) [reply NuGet.config:14]: The line isn't in NuGet.config here:

https://github.com/xamarin/xamarin-macios/blob/c272040bdb3aa988c48864183500487c404713b3/NuGet.config#L6-L12

It looks like I created this branch from an earlier commit, which doesn't have the removal, so it appears duplicated.

Comment (rolfbjarne) [reply NuGet.config:12]: I clicked the "Update branch" button (to merge with the tip of main), and it's no longer duplicated.

**Review by mandel-macaque — APPROVED** [review]

**Comment by rolfbjarne** [comment]
Test failures are unrelated. All different variations of:

* https://github.com/xamarin/maccore/issues/2494
* https://github.com/xamarin/maccore/issues/2493
* https://github.com/xamarin/maccore/issues/2491
* https://github.com/xamarin/maccore/issues/2443

**Pull request merged**